### PR TITLE
Shading AAP lib source code inside cosmos encryption project

### PR DESCRIPTION
**Pull request by simplynaveen20**
AAP libs Cryptography and AzureKeyVaultProvider are not yet GAed on Maven.
We decided to add their source code inside our encryption lib , so we can move forward with our GA plan.
Once we have AAP libs into maven we will revert this change and add the dependency back.
.NET encryption also did the same thing https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2967.
Every thing inside com.azure.cosmos.encryption.mdesrc is the code from AAP libs

This PR will wait on work item https://github.com/Azure/azure-cosmos-dotnet-v3/pull/2906 which is inflight,  which adds an abstraction layer to hide the underlying usage of Microsoft Data Encryption(MDE) library. Creating this to get the ball rolling for discussion.

### CODE REVIEW

File: sdk/parents/azure-client-sdk-parent/pom.xml [hunk 1]
```diff
@@ -629,7 +629,8 @@
               com.azure.core.test*:
               com.azure.endtoend*:
               com.azure.perf*:
-              com.azure.storage.internal.avro*
+              com.azure.storage.internal.avro*:
+			        com.azure.cosmos.encryption.mdesrc*
```
Comment (simplynaveen20) [inline sdk/parents/azure-client-sdk-parent/pom.xml:633]: Will fix this indentation

File: sdk/cosmos/azure-cosmos-encryption/pom.xml [hunk 2]
```diff
@@ -41,7 +41,8 @@ Licensed under the MIT License.
     <!-- CosmosSkip - This is not a module we want/expect external customers to consume. Skip breaking API checks. -->
     <!-- This can only be enabled once we release GA, as it needs a stable version to check for breaking changes. -->
     <revapi.skip>true</revapi.skip>
-
+    <!-- We are adding code for AAP jars, once AAP jar is GAed to maven, we will remove spotbugs.skip-->
+    <spotbugs.skip>true</spotbugs.skip>
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-encryption/pom.xml:45]: Will this disable spot bug check for every class under encryption? 
is there a way to only skip spot bug check for classes under com.azure.cosmos.encryption.mdesrc?

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

File: sdk/cosmos/azure-cosmos-encryption/pom.xml (hunk 2 above)
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos-encryption/pom.xml]: Done, added exclusion for aap source code in spotbug exclusion file

**Review by kushagraThapar — APPROVED** [review]
LGTM

**Pull request merged**